packages/forms-web-app/src/controllers/examination/email.js

```diff
@@ -2,24 +2,61 @@ const {
 	routesConfig: {
 		examination: {
 			directory: examinationDirectory,
-			pages: { nameMyself, email }
+			pages: { email }
 		}
 	}
 } = require('../../routes/config');
 
+const config = require('../../config');
+const examinationSessionStorage = config?.sessionStorage?.examination;
+
 const pageData = {
-	backLinkUrl: `${examinationDirectory + nameMyself.route}`,
 	id: email.id,
 	pageTitle: email.name,
-	title: email.name
+	title: email.name,
+	hint: "We'll use your email address to confirm we've received your submission. We will not publish your email address."
 };
 
 const getEmail = async (req, res) => {
+	const requestSession = req.session;
+	const userEmail = requestSession?.[examinationSessionStorage.name]?.userEmail;
+	const currentView = requestSession?.currentView;
+	userEmail && (pageData.email = userEmail);
+
+	if (!requestSession || !currentView || !currentView?.route)
+		return res.status(404).render('error/not-found');
+
+	pageData.backLinkUrl = `${examinationDirectory + currentView.route}`;
 	res.render(email.view, pageData);
 };
 
 const postEmail = async (req, res) => {
-	res.render(email.view, pageData);
+	const { body = {} } = req;
+	const { errors = {}, errorSummary = [] } = body;
+	const requestSession = req?.session;
+	const currentView = requestSession?.currentView;
+	const examinationStorage = requestSession?.[examinationSessionStorage.name];
```

Comment:
May be best to check for the examination session first, without it we should not carry on:

```
const examinationStorage = req.session[examinationSessionStorage.name];

if (!examinationStorage) ....
```

Reply:
It can but there is a caveat... With string interpolation we always guarantee JS will cast the variable values to a string. This doesn't happen with a sum (+) operator.

Eg:

`
const x = undefined;
const y = ' ';

console.log(typeof x + y) = undefined 
console.log(typeof `${x + y}`) = string
`

Reply:
Sorry, I'm not describing the issue so well. The first check we would want to do is to check for an 'examination' session as without this we can assume that the journey has not been started and we also don't have a key for our data. This is why it should be declared first and for the sake of optimisation, before any other logic.

Reply:
fixed it